src/Stratis.Bitcoin.IntegrationTests/BlockStore/BlockStoreSignaledTests.cs

```diff
@@ -0,0 +1,327 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Connection;
+using Stratis.Bitcoin.Features.BlockStore;
+using Stratis.Bitcoin.IntegrationTests.Common;
+using Stratis.Bitcoin.IntegrationTests.Common.EnvironmentMockUpHelpers;
+using Stratis.Bitcoin.P2P.Peer;
+using Stratis.Bitcoin.P2P.Protocol;
+using Stratis.Bitcoin.P2P.Protocol.Behaviors;
+using Stratis.Bitcoin.P2P.Protocol.Payloads;
+using Stratis.Bitcoin.Tests.Common;
+using Stratis.Bitcoin.Utilities;
+using Xunit;
+
+namespace Stratis.Bitcoin.IntegrationTests.BlockStore
+{
+    /// <summary>
+    /// Used for recording messages coming into a test node. Does not respond to them in any way.
+    /// </summary>
+    internal class TestBehavior : NetworkPeerBehavior
+    {
+        public Dictionary<string, List<IncomingMessage>> receivedMessageTracker = new Dictionary<string, List<IncomingMessage>>();
+
+        protected override void AttachCore()
+        {
+            this.AttachedPeer.MessageReceived.Register(this.OnMessageReceivedAsync);
+        }
+
+        protected override void DetachCore()
+        {
+            this.AttachedPeer.MessageReceived.Unregister(this.OnMessageReceivedAsync);
+        }
+
+        private async Task OnMessageReceivedAsync(INetworkPeer peer, IncomingMessage message)
+        {
+            try
+            {
+                await this.ProcessMessageAsync(peer, message).ConfigureAwait(false);
+            }
+            catch (OperationCanceledException)
+            {
+            }
+        }
+
+        private async Task ProcessMessageAsync(INetworkPeer peer, IncomingMessage message)
+        {
+            if (!this.receivedMessageTracker.ContainsKey(message.Message.Payload.Command))
+                this.receivedMessageTracker[message.Message.Payload.Command] = new List<IncomingMessage>();
+
+            this.receivedMessageTracker[message.Message.Payload.Command].Add(message);
+        }
+
+        public override object Clone()
+        {
+            var res = new TestBehavior();
+
+            return res;
+        }
+    }
+
+    public class BlockStoreSignaledTests
+    {
+        protected readonly ILoggerFactory loggerFactory;
+        private readonly Network network;
+
+        public BlockStoreSignaledTests()
+        {
+            this.loggerFactory = new LoggerFactory();
+
+            this.network = KnownNetworks.RegTest;
```

Comment:
Can you rather use new BitcoinRegTest here? We want to move away from the static container eventually.